pandas/tests/io/excel/test_style.py

```diff
@@ -333,3 +336,39 @@ def test_styler_to_s3(s3_public_bucket, s3so):
             f"s3://{mock_bucket_name}/{target_file}", index_col=0, storage_options=s3so
         )
         tm.assert_frame_equal(result, df)
+
+
+def test_format_hierarchical_rows_periodindex_merge_cells():
+    # GH#60099
+    df = DataFrame(
+        {"A": [1, 2]},
+        index=MultiIndex.from_arrays(
+            [period_range("2023-01", "2023-02", freq="M"), ["X", "Y"]],
+            names=["date", "category"],
+        ),
+    )
+    formatter = ExcelFormatter(df, merge_cells=True)
+    formatted_cells = list(formatter._format_hierarchical_rows())
+
+    for cell in formatted_cells:
+        assert not isinstance(
+            cell.val, Period
+        ), "Period should be converted to Timestamp"
+
+
+def test_format_hierarchical_rows_periodindex_no_merge_cells():
```

Comment:
Can you combine this test with the one above using `@pytest.mark.parametrize`

Reply:
Sure. I'll modify it soon. Thanks a lot for comments!